build_tools/cmake/iree_plugin_register.cmake

```diff
@@ -0,0 +1,133 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Ids of all plugins that we have statically linked.
+set_property(GLOBAL PROPERTY IREE_COMPILER_LINKED_PLUGIN_IDS "")
+# Libraries to add to the plugin registry for all statically linked plugins.
+set_property(GLOBAL PROPERTY IREE_COMPILER_LINKED_PLUGIN_LIBS "")
+
+# iree_compiler_register_plugin()
+# Within a plugin package, registers the plugin by id with the build system,
+# associating it with a registration target.
+function(iree_compiler_register_plugin)
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "PACKAGE;PLUGIN_ID;TARGET"
+    ""
+    ${ARGN}
+  )
+
+  # Prefix the library with the package name, so we get: iree_package_name.
+  if(_RULE_PACKAGE)
+    set(_PACKAGE_NS "${_RULE_PACKAGE}")
+    string(REPLACE "::" "_" _PACKAGE_NAME ${_RULE_PACKAGE})
+  else()
+    iree_package_ns(_PACKAGE_NS)
+    iree_package_name(_PACKAGE_NAME)
+  endif()
+
+  # Replace target passed by ::name with iree::package::name
+  list(TRANSFORM _RULE_TARGET REPLACE "^::" "${_PACKAGE_NS}::")
+
+  # TODO: Can have more control on what gets linked.
+  message(STATUS "Registering static linked compiler plugin '${_RULE_PLUGIN_ID}' (${_RULE_TARGET})")
+  set_property(GLOBAL APPEND PROPERTY IREE_COMPILER_LINKED_PLUGIN_IDS "${_RULE_PLUGIN_ID}")
+  set_property(GLOBAL APPEND PROPERTY IREE_COMPILER_LINKED_PLUGIN_LIBS "${_RULE_TARGET}")
```

Comment:
What sort of additional control would be useful?

(also, TODO style: could add a username or issue number)

Reply:
CMake didn't detect the rename of this file and this was pre-existing. But I'll remove it. I almost certainly wrote it and couldn't tell you what it means :)